### PR TITLE
Logging sample

### PR DESCRIPTION
**Pull request by shahzaibj**

### CODE REVIEW

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java [hunk 1]
```diff
@@ -40,27 +46,39 @@
...
             @Override
             public void log(String tag, Logger.LogLevel logLevel, String message, boolean containsPII) {
-                // contains PII indicates that if the log message contains PII information. If Pii logging is
-                // disabled, the sdk never returns back logs with Pii.
-                mLogSize = mLogs.toString().getBytes().length;
-                if (mLogSize + message.getBytes().length >= LOG_SIZE) {
-                    clearLogs();
-                }
-                mLogs.append(message).append('\n');
+                writeToLogFile(getApplicationContext(), "TAG: " + message);
```
Comment (shoatman) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java]: Looks like we didn't log the tag value.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java [hunk 2]
```diff
@@ -40,27 +46,39 @@
...
+    private synchronized void writeToLogFile(Context ctx, String msg) {
+        File directory = ctx.getDir(ctx.getPackageName(), Context.MODE_PRIVATE);
+        File logFile = new File(directory, "msallogs.txt");
+        FileOutputStream outputStream = null;
+        try {
+            outputStream = new FileOutputStream(logFile, true);
+            OutputStreamWriter osw = new OutputStreamWriter(outputStream);
+            osw.write(msg);
+            osw.flush();
+            osw.close();
```
Comment (shoatman) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java]: Should we have a finally block for closing the stream.

File: testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java [hunk 3]
```diff
@@ -40,27 +46,39 @@
...
-    private static final int LOG_SIZE = 1024 * 1024;
     private StringBuilder mLogs;
-    private int mLogSize;
+
+    private synchronized void writeToLogFile(Context ctx, String msg) {
+        File directory = ctx.getDir(ctx.getPackageName(), Context.MODE_PRIVATE);
+        File logFile = new File(directory, "msallogs.txt");
+        FileOutputStream outputStream = null;
+        try {
+            outputStream = new FileOutputStream(logFile, true);
```
Comment (shoatman) [inline testapps/testapp/src/main/java/com/microsoft/identity/client/testapp/MsalSampleApp.java]: Are going to rollover this file when it becomes too large?  I'd be concerned about someone taking this code as is and copying it in to their application.

**Comment by shahzaibj** [comment]
@shoatman This is just a quick sample I made for a customer that wanted to know how to use logger and callbacks. The primary motive here was to demonstrate how the callback works and I am not concerned about the specifics of writing to file. I am not actually planning on taking this change and going to close this PR. The test app already has a basic implementation of logger but if you think we need a more robust sample then I think we can create one on the README page or perhaps in the Android Azure Sample app. Let me know your thoughts.

**Pull request closed without merging**